orchagent/intfsorch.cpp

```diff
@@ -712,21 +712,24 @@ void IntfsOrch::doTask(Consumer &consumer)
                     continue;
                 }
 
-                /* Set nat zone id */
-                if ((!nat_zone.empty()) and (port.m_nat_zone_id != nat_zone_id))
+                if (gPortsOrch->getPort(alias, port))
```

Comment:
this `gPortsOrch->getPort(alias, port))`check is already done above( line 665). Why do we need to do the check here again ?

Reply:
@arlakshm RIF objects are created in setIntf() before calling set_nat_zone API and the RIF object handle is set in port.m_rif_id. To get the Port object with rif handle, getPort() is invoked before setting the NAT zone.